framework/validators/AtLeastOneValidator.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\validators;
+
+use Yii;
+use yii\base\InvalidConfigException;
+
+/**
+ * AtLeastValidator checks if at least one of many attributes are filled.
+ *
+ * In the following example, the `attr1`, `attr2` and `attr3` attributes will
+ * be verified. If at least one of them are not filled, all will receive an error:
+ *
+ * ```php
+ *  ['attr1', 'atLeastOne', 'alternativeAttributes' => ['attr2', 'attr3']],
+ * ```
+ *
+ * In the following example, the `attr1`, `attr2` and `attr3` attributes will
+ * be verified. If at least two of them are not filled, `attr1` and `attr3` will
+ * receive an error:
+ *
+ * ```php
+ *  ['attr1', 'atLeastOne', 'alternativeAttributes' => ['attr2', 'attr3'], 'min' => 2, 'errorAttributes' => ['attr1', 'attr3']],
+ * ```
+ *
+ * Important: Only one attribute should be specified in the attributes.
+ * The additional ones must go into [[alternativeAttributes]] param.
+ *
+ * The attributes that will receive the error message can be specified into [[errorAttributes]]
+ * param. If none is specified, all envolved attributes will receive the error.
+ *
+ * @author Sidney Lins <slinstj@gmail.com>
+ * @since 2.0.14
+ */
+class AtLeastOneValidator extends Validator
+{
```

Comment:
I'd rename this class to something more specific. Something like `RequireGroupValidator`

Reply:
Group validator doesn't sound clear to me. Suggests nothing about behavior.

Reply:
its says someting about a group of attributes that are required...